[BUG] Crash while "Turn off work apps"

### Describe the bug

Go app list , click on work apps tab,trunk off work apps

### Steps to reproduce

Steps to reproduce the behavior:
1. Go to 'App drawer'
2. Click on 'work' tab
3. Click on 'turn off work app'
4. See the error


### Expected behavior

Trun off the work apps

### Screenshots

_No response_

### Device information

- Device: Asus Zenfone 8 
- OS: Stock android 13 beta
- App version: 12.1.0 Alpha 4


### Additional context

https://cdn.discordapp.com/attachments/858972718611562496/1048122897954263073/Lawnchair_bug_report_Dec_2_2022_22511_PM.txt

```
java.lang.IllegalArgumentException: User 2357 is not a profile
    at android.os.Parcel.createExceptionOrNull(Parcel.java:3015)
    at android.os.Parcel.createException(Parcel.java:2995)
    at android.os.Parcel.readException(Parcel.java:2978)
    at android.os.Parcel.readException(Parcel.java:2920)
    at android.os.IUserManager$Stub$Proxy.requestQuietModeEnabled(IUserManager.java:3210)
    at android.os.UserManager.requestQuietModeEnabled(UserManager.java:4565)
    at android.os.UserManager.requestQuietModeEnabled(UserManager.java:4550)
    at android.os.UserManager.requestQuietModeEnabled(UserManager.java:4517)
    at com.android.launcher3.allapps.WorkProfileManager.lambda$setWorkProfileEnabled$0(WorkProfileManager.java:3)
    at com.android.launcher3.allapps.WorkProfileManager.a(Unknown Source:0)
    at com.android.launcher3.allapps.o.run(Unknown Source:4)
    at android.os.Handler.handleCallback(Handler.java:942)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loopOnce(Looper.java:241)
    at android.os.Looper.loop(Looper.java:358)
    at android.os.HandlerThread.run(HandlerThread.java:67)
Caused by: android.os.RemoteException: Remote stack trace:
    at com.android.server.pm.UserManagerService.setQuietModeEnabled(UserManagerService.java:1235)
    at com.android.server.pm.UserManagerService.requestQuietModeEnabled(UserManagerService.java:1143)
    at android.os.IUserManager$Stub.onTransact(IUserManager.java:1662)
    at android.os.Binder.execTransactInternal(Binder.java:1280)
    at android.os.Binder.execTransact(Binder.java:1244)
```

On Pixel Android 12:

```
java.lang.SecurityException: Can't modify quiet mode, caller is neither foreground default launcher nor has MANAGE_USERS/MODIFY_QUIET_MODE permission
    at android.os.Parcel.createExceptionOrNull(Parcel.java:2425)
    at android.os.Parcel.createException(Parcel.java:2409)
    at android.os.Parcel.readException(Parcel.java:2392)
    at android.os.Parcel.readException(Parcel.java:2334)
    at android.os.IUserManager$Stub$Proxy.requestQuietModeEnabled(IUserManager.java:3583)
    at android.os.UserManager.requestQuietModeEnabled(UserManager.java:3967)
    at android.os.UserManager.requestQuietModeEnabled(UserManager.java:3952)
    at android.os.UserManager.requestQuietModeEnabled(UserManager.java:3919)
    at com.android.launcher3.allapps.WorkProfileManager.lambda$setWorkProfileEnabled$0$com-android-launcher3-allapps-WorkProfileManager(WorkProfileManager.java:97)
    at com.android.launcher3.allapps.WorkProfileManager$$ExternalSyntheticLambda0.run(Unknown Source:4)
    at android.os.Handler.handleCallback(Handler.java:938)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loopOnce(Looper.java:201)
    at android.os.Looper.loop(Looper.java:288)
    at android.os.HandlerThread.run(HandlerThread.java:67)
Caused by: android.os.RemoteException: Remote stack trace:
    at com.android.server.pm.UserManagerService.ensureCanModifyQuietMode(UserManagerService.java:1168)
    at com.android.server.pm.UserManagerService.requestQuietModeEnabled(UserManagerService.java:1075)
    at android.os.IUserManager$Stub.onTransact(IUserManager.java:1648)
    at android.os.Binder.execTransactInternal(Binder.java:1179)
    at android.os.Binder.execTransact(Binder.java:1143)
```

--- a/src/com/android/launcher3/allapps/WorkProfileManager.java
+++ b/src/com/android/launcher3/allapps/WorkProfileManager.java
@@ -94,7 +94,12 @@ public class WorkProfileManager implements PersonalWorkSlidingTabStrip.OnActiveP
                 if (Process.myUserHandle().equals(userProfile)) {
                     continue;
                 }
-                mUserManager.requestQuietModeEnabled(!enabled, userProfile);
+                // https://github.com/LawnchairLauncher/lawnchair/issues/3145
+                try {
+                    mUserManager.requestQuietModeEnabled(!enabled, userProfile);
+                } catch (RuntimeException e) {
+                    Log.e(TAG, "Failed to set quiet mode for user " + userProfile, e);
+                }
             }
         });
     }

--- a/src/com/android/launcher3/allapps/WorkProfileManager.java
+++ b/src/com/android/launcher3/allapps/WorkProfileManager.java
@@ -26,6 +26,7 @@ import android.os.Process;
 import android.os.UserHandle;
 import android.os.UserManager;
 import android.util.Log;
+import android.widget.Toast;
 
 import androidx.annotation.IntDef;
 import androidx.annotation.Nullable;
@@ -98,7 +99,7 @@ public class WorkProfileManager implements PersonalWorkSlidingTabStrip.OnActiveP
                 try {
                     mUserManager.requestQuietModeEnabled(!enabled, userProfile);
                 } catch (RuntimeException e) {
-                    Log.e(TAG, "Failed to set quiet mode for user " + userProfile, e);
+                    Toast.makeText(mAllApps.getContext(), e.getMessage(), Toast.LENGTH_SHORT).show();
                 }
             }
         });

